### PR TITLE
Adds tests to the ECM class.

### LINKED ISSUE #20: Add tests for Entity and EntityComponentManager
### Summary

- [ ] Add tests for entity https://github.com/maliput/maliput_sim/blob/9ea953bd9034fea3a6a74515cb58689b6c6155a8/maliput_sim/core/ecm.py#L34

- [ ] Add tests for ecm https://github.com/maliput/maliput_sim/blob/9ea953bd9034fea3a6a74515cb58689b6c6155a8/maliput_sim/core/ecm.py#L75

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Closes #20 

## Summary
What the title says

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: test/core/ecm_test.py [hunk 1]
```diff
@@ -90,3 +90,98 @@ class ComponentB(MagicMock):
             }
         }
         self.assertEqual(self.entity.get_state(), expected_state)
+
+
+class TestEntityComponentManager(unittest.TestCase):
+
+    def setUp(self):
+        self.ecm = EntityComponentManager()
```
Comment (agalbachicar) [inline test/core/ecm_test.py:98]: What about `dut`?

File: test/core/ecm_test.py [hunk 2]
```diff
@@ -90,3 +90,98 @@ class ComponentB(MagicMock):
...
+        entity2 = self.ecm.create_entity()
+
+        retrieved_entity1 = self.ecm.get_entity(entity1._entity_id)
+        retrieved_entity2 = self.ecm.get_entity(entity2._entity_id)
+
+        self.assertEqual(entity1, retrieved_entity1)
+        self.assertEqual(entity2, retrieved_entity2)
+
+    def test_get_entities_with_component(self):
+        component_type = MagicMock()
```
Comment (agalbachicar) [inline test/core/ecm_test.py:128]: Shouldn't this be the type?

File: test/core/ecm_test.py [hunk 3]
```diff
@@ -90,3 +90,98 @@ class ComponentB(MagicMock):
...
+        print(entity2.get_components())
+        print(entity3.get_components())
+        entities_with_component = self.ecm.get_entities_with_component(component_type)
+        self.assertEqual(len(entities_with_component), 2)
+        self.assertIn(entity1, entities_with_component)
+        self.assertIn(entity3, entities_with_component)
+
+    def test_get_entities_of_type(self):
+        entity_type = "TestType"
+        different_entity_type = "DifferentType"
```
Comment (agalbachicar) [inline test/core/ecm_test.py:146]: Move this next to line 147 so it is close to where it is used.

**Review by agalbachicar — CHANGES_REQUESTED** [review]
Overall, it LGTM
Minor comments to go.

File: test/core/ecm_test.py (hunk 3 above)
Comment (francocipollone) [reply test/core/ecm_test.py:146]: Done.

File: test/core/ecm_test.py (hunk 2 above)
Comment (francocipollone) [reply test/core/ecm_test.py:128]: Yes it should. Done.

File: test/core/ecm_test.py (hunk 1 above)
Comment (francocipollone) [reply test/core/ecm_test.py:98]: Done.

**Pull request merged**